nasa/harmony
# Harmony 1764 2

## Jira Issue ID
HARMONY-1764

## Description
We use SQS queues to communicate between the main harmony executable and internal services. SQS queue messages are limited to 256K. When a user specifies many variables the data operation can exceed this size. This change uses an in-memory cache for the data operation so it does not need to be passed on the queues. 

## Local Test Steps
Testing this change involves a lot of steps.
1. Check out this branch
2. Run the normal dev services and test with a standard query
3. Rebuild the service images
```
npm run build-all
```
4. Change your .env to run harmony in a box
5. Run harmony in a box
```
./bin/bootstrap-harmony
```
6. Run some queries to test things out

To test out a full request with many variables you will need to rebuild a service (like trajectory-subsetter) with my harmony-1764-2 branch of harmony-service-lib-py. I can provide a query for that if needed.

## PR Acceptance Checklist
